extensions/wikia/HTTPSSupport/tests/HTTPSSupportHooksTest.php

```diff
@@ -86,4 +86,78 @@ public function onBeforeResourceLoaderCSSMinifierDataProvider() {
 			]
 		];
 	}
+
+	public function testOnLinkerMakeExternalLink() {
+
+		$exampleString = '';
+		$exampleBool = false;
+		$exampleArray = [];
+
+		#Should not be upgraded to HTTPS
+
+		#external URL
+		$url = 'http://www.example.com';
+		$expectedResult = 'http://www.example.com';
+		HTTPSSupportHooks::onLinkerMakeExternalLink($url,$exampleString , $exampleBool, $exampleArray);
+		$this->assertEquals( $expectedResult, $url );
+
+		#multiple subdomain wikia.com
+		$url = 'http://ja.starwars.wikia.com';
+		$expectedResult = 'http://ja.starwars.wikia.com';
+		HTTPSSupportHooks::onLinkerMakeExternalLink($url,$exampleString , $exampleBool, $exampleArray);
+		$this->assertEquals( $expectedResult, $url );
+
+
+		#Should be upgraded to HTTPS
+
+		#single subdomain wikia.com
+		$url = 'http://starwars.wikia.com';
+		$expectedResult = 'https://starwars.wikia.com';
+		HTTPSSupportHooks::onLinkerMakeExternalLink($url,$exampleString , $exampleBool, $exampleArray);
+		$this->assertEquals( $expectedResult, $url );
+
+		#wikia.org
+		$url = 'http://starwars.wikia.com';
+		$expectedResult = 'https://starwars.wikia.com';
+		HTTPSSupportHooks::onLinkerMakeExternalLink($url,$exampleString , $exampleBool, $exampleArray);
+		$this->assertEquals( $expectedResult, $url );
+
+		#fandom.com
+		$url = 'http://starwars.fandom.com';
+		$expectedResult = 'https://starwars.fandom.com';
+		HTTPSSupportHooks::onLinkerMakeExternalLink($url,$exampleString , $exampleBool, $exampleArray);
+		$this->assertEquals( $expectedResult, $url );
+
+		#verify.fandom.com
+		$url = 'http://starwars.verify.fandom.com';
```

Comment:
Should we call `$this->mockEnvironment( $env );` to fake the env before calling `onLinkerMakeExternalLink`?